src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,94 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [title, setTitleField] = useState('');
+  const [description, setDescriptionField] = useState('');
+  const [imgUrl, setImageUrlField] = useState('');
+  const [imdbUrl, setImdbUrlField] = useState('');
+  const [imdbId, setImdbIdField] = useState('');
```

Comment:
It's not advisable to generate such a large number of states when we only have one instance (movie). 
It'd be much better to utilize a single state with an object inside.

1. Create a variable outside the Component
```
const initialMovieState = {
  title: '',
  description: '',
  imgUrl: '',
  imdbUrl: '',
  imdbId: ''
}
```

2. Use one state for movie fields
3. Pass `initialMovieState` object when you invoke reset `function`
4. Whenever you need to change value of any field you should pass `event` and then just change a value of a movie object by key. In this case you will reuse one handler instead of handlers for each input
Example: 

```
const handleInputChange = (e: ...) => {
  const {name, value} = e.target; setMovie(prevInputs => ({...prevInputs, [name]: value}))
}
```

5. for reset function you will just pass initialMovieState


```suggestion
  const [movie, setMovie] = useState(initialMovieState);
```